File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    dic = ['A', 'E', 'I', 'O', 'U']
    found = False
    
    def dfs(cur):
        nonlocal answer, found
        if found: return
        if cur != "":
            answer += 1
            if cur == word:
                found = True
                return
        
        if len(cur) == 5: return
    
        for d in dic:
            dfs(cur + d)
        
    dfs("")
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.swift
import Foundation

func solution(_ word:String) -> Int {
    
    var answer = 0
    var find = false
    
    func dfs(_ cur: String) {
        if find { return }
        if !cur.isEmpty {
            answer += 1
            if cur == word { 
                find = true
                return 
            }
        }
        if cur.count == 5 { return }
        
        for w in ["A", "E", "I", "O", "U"] {
            dfs(cur+w)
        }
    }
    
    dfs("")
    
    return answer
}